### PR TITLE
minor updates

### PR DESCRIPTION
**Pull request by pahumadad**
* add the `--version` `CLI` argument to show the `Micro` installed version
* update the `Params` class to check `Micro` params and set them as internal environment variables
* check required arguments and show the `CLI` help when they are not set
* replace all `raise` to show errors by print errors to `stderr` and `exit(1)
* update unit tests

### CODE REVIEW

File: micro/core/config.py [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
 
     def key(self, name):
         return self.__conf.get(name)
 
-    def __load(self):
-        with open(self.__path) as conf_file:
+    def __load(self, path):
+        if not os.path.exists(path) or os.path.isdir(path):
+            sys.stderr.write("ERROR: config file not found: {}".format(path))
+            sys.exit(1)
```
Comment (fortizc) [inline micro/core/config.py]: A better way to do this is `sys.exit("ERROR: config file not found: {}".format(path))` see https://docs.python.org/3/library/sys.html#sys.exit

File: micro/core/params.py [hunk 2]
```diff
@@ -39,34 +93,66 @@ def __priority_param(self, cli_param, env_var_name, config_key):
...
+    def __set_all(self):
+        for p in PARAMS:
+            value = self.__priority_param(self.__args.get(PARAMS[p]["var"]),
+                                          PARAMS[p]["env"],
+                                          PARAMS[p]["var"])
+            if value:
+                envvar = "_" + PARAMS[p]["env"]
+                os.environ[envvar] = str(value)
+
+    def __check_required(self):
```
Comment (fortizc) [inline micro/core/params.py:107]: Here the previous comment is valid too ;)

File: micro/core/utils.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import os
+import sys
+
+
+def set_folder(path):
+    try:
+        os.makedirs(path, exist_ok=True)
+    except PermissionError:
```
Comment (fortizc) [inline micro/core/utils.py:8]: Here again, and in all the next "exits"

File: setup.py [hunk 4]
```diff
@@ -1,23 +1,40 @@
...
+        return f.read()
+
+
+def find_version(*file_paths):
+    version_file = read(*file_paths)
+    version_match = re.search(
+        r"^__version__ = ['\"]([^'\"]*)['\"]",
+        version_file,
+        re.M,
+    )
```
Comment (fortizc) [inline setup.py]: Is necessary the last comma? for the way how you indent this code and this last comma, it seems to be a tuple.
Now in a very personal comment, I prefer this way:
```python
version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]",
                          version_file,
                          re.M)
```

File: micro/core/config.py (hunk 1 above)
Comment (pahumadad) [reply micro/core/config.py]: Oh, and results in an exit code `1` and the text is sent to `stderr`..., so it is a much better way to do it, great! 👍

Comment (pahumadad) [reply micro/core/config.py]: solved in commit b2f08aba5dff0c10a7c217c824e0086e5472c060

File: micro/core/params.py (hunk 2 above)
Comment (pahumadad) [reply micro/core/params.py:107]: solved in commit b2f08aba5dff0c10a7c217c824e0086e5472c060

File: micro/core/utils.py (hunk 3 above)
Comment (pahumadad) [reply micro/core/utils.py:8]: solved in commit b2f08aba5dff0c10a7c217c824e0086e5472c060

File: setup.py (hunk 4 above)
Comment (pahumadad) [reply setup.py]: solved in commit b2f08aba5dff0c10a7c217c824e0086e5472c060

**Review by fortizc — APPROVED** [review]

**Pull request merged**